picasso/src/main/java/com/squareup/picasso3/Action.kt

```diff
@@ -16,19 +16,20 @@
 package com.squareup.picasso3
 
 import com.squareup.picasso3.RequestHandler.Result
+import java.lang.ref.WeakReference
 
-internal abstract class Action(
+internal abstract class Action<T>(
   val picasso: Picasso,
   val request: Request
 ) {
   var willReplay = false
   var cancelled = false
 
+  abstract val target: WeakReference<T>
```

Comment:
Consider going one step further and making WeakReference an implementation detail of target, and target final.

Something like this:

```kotlin
internal abstract class Action<T>(
  val picasso: Picasso,
  val request: Request
  target: T
) {

  private val targetRef = WeakReference(target)
  
  val target: T?
    get() = targetRef.get()

}

```

Reply:
Good idea! Thanks @pyricau 🙏🏻

Reply:
Though if I go this route, `FetchAction` and `GetAction` get a little weird because they don't necessarily want callers to provide a target

Reply:
@jrodbx &

Reply:
Resolved!